src/toil_scripts/tools/preprocessing.py

```diff
@@ -131,9 +228,63 @@ def run_gatk_preprocessing(job, cores, bam, bai, ref, ref_dict, fai, phase, mill
+
+    :param JobFunctionWrappingJob job: passed automatically by Toil
+    :param int cores: Maximum number of cores on a worker node
+    :param str bam: Sample BAM FileStoreID
+    :param str bai: Bam Index FileStoreID
+    :param str ref: Reference genome FileStoreID
+    :param str ref_dict: Reference dictionary FileStoreID
+    :param str fai: Reference index FileStoreID
+    :param str phase: Phase VCF FileStoreID
+    :param str mills: Mills VCF FileStoreID
+    :param str dbsnp: DBSNP VCF FileStoreID
+    :param str mem: Memory value to be passed to children. Needed for CI tests
+    :param bool unsafe: If True, runs gatk UNSAFE mode: "-U ALLOW_SEQ_DICT_INCOMPATIBILITY"
+    :return: BAM and BAI FileStoreIDs from Print Reads
+    :rtype: tuple(str, str)
+    """
+    rm_secondary = job.wrapJobFn(samtools_view, bam, flag='0x800', mock=mock)
+    picard_sort = job.wrapJobFn(picard_sort_sam, rm_secondary.rv(), xmx=mem, mock=mock)
+    # MarkDuplicates runs best when Xmx <= 10G
+    mdups_mem = '{}G'.format(min(10, int(mem[:-1])))
+    mdups = job.wrapJobFn(picard_mark_duplicates, picard_sort.rv(0), picard_sort.rv(1),
+                          xmx=mdups_mem, mock=mock)
+    realigner_target = job.wrapJobFn(run_realigner_target_creator, cores, mdups.rv(0), mdups.rv(1),
+                                     ref, ref_dict, fai, phase, mills, mem, unsafe=unsafe,
+                                     mock=mock)
+    indel_realign = job.wrapJobFn(run_indel_realignment, realigner_target.rv(), bam, bai, ref,
+                                  ref_dict, fai, phase, mills, mem, unsafe=unsafe, mock=mock)
+    base_recal_one = job.wrapJobFn(run_base_recalibration, cores, indel_realign.rv(0),
+                                   indel_realign.rv(1), ref, ref_dict, fai, dbsnp, mills, mem,
+                                   unsafe=unsafe, mock=mock)
+    base_recal_two = job.wrapJobFn(run_base_recalibration, cores, indel_realign.rv(0),
+                                   indel_realign.rv(1), ref, ref_dict, fai, dbsnp, mills, mem,
+                                   unsafe=unsafe, mock=mock, bqsr=base_recal_one.rv())
+    # TODO Add AnalyzeCovariates function
+    print_reads = job.wrapJobFn(run_print_reads, cores, base_recal_two.rv(), indel_realign.rv(0),
+                                indel_realign.rv(1), ref, ref_dict, fai, mem, unsafe=unsafe,
+                                mock=mock)
+
+    job.addChild(rm_secondary)
+    rm_secondary.addChild(picard_sort)
+    picard_sort.addChild(mdups)
+    mdups.addChild(realigner_target)
+    realigner_target.addChild(indel_realign)
+    indel_realign.addChild(base_recal_one)
+    base_recal_one.addChild(base_recal_two)
+    base_recal_two.addChild(print_reads)
+    return print_reads.rv(0), print_reads.rv(1)
+
+
+def run_realigner_target_creator(job, cores, bam, bai, ref, ref_dict, fai, phase, mills, mem,
+                                 unsafe=False, mock=False):
```

Comment:
Document `mock`. Split params out to one per line unless there is a logical grouping.